File: atomic-x/android/src/main/kotlin/io/trtc/tuikit/atomicx/videorecorder/VideoRecorderPlugin.kt
package io.trtc.tuikit.atomicx.videorecorder

import android.util.Log
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class VideoRecorderPlugin(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) : 
    MethodChannel.MethodCallHandler {
    
    companion object {
        private const val TAG = "VideoRecorderPlugin"
        private const val METHOD_CHANNEL_NAME = "atomic_x/video_recorder"
    }

    private val methodChannel: MethodChannel = MethodChannel(
        flutterPluginBinding.binaryMessenger,
        METHOD_CHANNEL_NAME
    )
    
    private var videoRecorderHandler: VideoRecorderHandler? = null

    init {
        methodChannel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "startRecord" -> {
                if (videoRecorderHandler == null) {
                    videoRecorderHandler = VideoRecorderHandler()
                }
                videoRecorderHandler?.handleStartRecord(call, result)
            }
            else -> {
                result.notImplemented()
            }
        }
    }

    fun dispose() {
        methodChannel.setMethodCallHandler(null)
        videoRecorderHandler = null
    }
}

File: atomic-x/ios/Classes/VideoRecorder/VideoRecorderPlugin.swift
import Flutter
import Foundation
import UIKit

class VideoRecorderPlugin: NSObject {
    private var methodChannel: FlutterMethodChannel
    private var videoRecorderHandler: VideoRecorderHandler
    
    init(registrar: FlutterPluginRegistrar) {
        self.methodChannel = FlutterMethodChannel(
            name: "atomic_x/video_recorder",
            binaryMessenger: registrar.messenger()
        )
        
        // 获取根视图控制器
        var rootViewController: UIViewController?
        if #available(iOS 13.0, *) {
            rootViewController = UIApplication.shared.windows.first?.rootViewController
        } else {
            rootViewController = UIApplication.shared.keyWindow?.rootViewController
        }
        
        self.videoRecorderHandler = VideoRecorderHandler(viewController: rootViewController)
        
        super.init()
        
        registrar.addMethodCallDelegate(self, channel: methodChannel)
    }
    
    func dispose() {
        methodChannel.setMethodCallHandler(nil)
    }
}

extension VideoRecorderPlugin: FlutterPlugin {
    static func register(with registrar: FlutterPluginRegistrar) {
        // This is handled in the main plugin
    }
}

extension VideoRecorderPlugin {
    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "startRecord":
            videoRecorderHandler.handleStartRecord(call: call, result: result)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
